mars/scheduler/graph.py

```diff
@@ -625,33 +653,11 @@ def create_operand_actors(self, _clean_io_meta=True):
 
             op_info = operand_infos[op_key]
 
-            # collect operand i/o information
-            predecessor_keys = set()
-            successor_keys = set()
-            input_chunk_keys = set()
-            shared_input_chunk_keys = set()
-            chunks = set()
-
-            for c in self._op_key_to_chunk[op_key]:
-                for pn in chunk_graph.iter_predecessors(c):
-                    predecessor_keys.add(pn.op.key)
-                    input_chunk_keys.add(pn.key)
-                    if chunk_graph.count_successors(pn) > 1:
-                        shared_input_chunk_keys.add(pn.key)
-                successor_keys.update(pn.op.key for pn in chunk_graph.iter_successors(c))
-                chunks.update(co.key for co in c.op.outputs)
-
-            io_meta = dict(
-                predecessors=list(predecessor_keys),
-                successors=list(successor_keys),
-                input_chunks=list(input_chunk_keys),
-                shared_input_chunks=list(shared_input_chunk_keys),
-                chunks=list(chunks),
-            )
+            io_meta = self._collect_operand_io_meta(chunk_graph, self._op_key_to_chunk[op_key])
             op_info['op_name'] = op_name
             op_info['io_meta'] = io_meta
 
-            if predecessor_keys:
+            if io_meta['predecessors']:
                 state = 'UNSCHEDULED'
```

Comment:
This can be changed into real enum value instead of names and values to reduce code complexity.

Reply:
Done.